src/Autocomplete/Autocomplete.tsx

```diff
@@ -18,9 +18,11 @@ import { isKeyValueOption } from "../utils/common";
 
 export default function Autocomplete<
   Value extends KeyValueOption | string,
-  Multiple extends boolean | undefined
+  Multiple extends boolean | undefined,
+  DisableClearable extends boolean | undefined
 >({
   defaultValue,
+  disableClearable,
```

Comment:
should this not have a default, presumably false so we don't break any existing behaviour.

Reply:
Typescript won't allow me to do this. The DisableClearable is a type parameter with a constraint on it. The only thing it will allow you to assign is undefined. It's identical to the Multiple type and prop and produces the same error.

Reply:
I'm a little bit confused why we need to be using generics for our own wrapper component if I'm honest. I can remove them and then get the behaviour I think we should have but maybe I'm missing something here. Maybe worth a quick call with @mattcorner and the two of us as I think he may have first developed the type definitions for this component.

Reply:
It looks like the generic affects the type of various different props, and using the AutocompleteProps type from MUI  requires the parameters. I'm getting a few errors pop up when I try to remove them.

From @mattcorner , "the generic affects some of the other props, so for instance if multiple is true, then onChange accepts an array, otherwise it accepts a scalar".

Still happy to talk further, but it feels like it should be separate to this PR?

Reply:
Yes we should use generic when we derive props from autocomplete and just pass them on as our own but that doesn't mean our own component needs to use generics? I have some code stashed locally which I think works so we can have a call when you are both free. I'm ok to handle it elsewhere, as long as the component still has the same defaults as the current component.